lib/msal-node/test/cache/Storage.spec.ts

```diff
@@ -0,0 +1,161 @@
+import { LogLevel, Logger, AccountEntity, CacheManager, AccessTokenEntity } from '@azure/msal-common';
+import { JsonCache, InMemoryCache } from './../../src/cache/serializer/SerializerTypes';
+import { Deserializer } from './../../src/cache/serializer/Deserializer';
+import { Storage } from './../../src/cache/Storage';
+import debug from 'debug';
+
+const cacheJson = require('./serializer/cache.json');
+
+describe("Storage tests for msal-node: ", () => {
+    let inMemoryCache: InMemoryCache = {
+        accounts: {},
+        idTokens: {},
+        accessTokens: {},
+        refreshTokens: {},
+        appMetadata: {},
+    };
+
+    let logger: Logger;
+
+
+    beforeEach(() => {
+        const cache = JSON.stringify(cacheJson);
+        const jsonCache: JsonCache = Deserializer.deserializeJSONBlob(cache);
+        inMemoryCache = Deserializer.deserializeAllCache(jsonCache);
+
+        const loggerOptions = {
+            loggerCallback: (
+                level: LogLevel,
+                message: string,
+                containsPii: boolean
+            ) => {
+                debug(`msal:${LogLevel[level]}${containsPii ? '-Pii' : ''}`)(message);
+            },
+            piiLoggingEnabled: false,
+            logLevel: LogLevel.Info,
+        };
+        logger = new Logger(loggerOptions!);
+    });
+
+    it("Constructor tests: ", () => {
+        const nodeStorage = new Storage(logger);
+        expect(nodeStorage).toBeInstanceOf(Storage);
+
+        const cache = nodeStorage.getCache();
+        expect(Object.keys(cache).length).toBe(0);
+
+        const inMemoryCache = nodeStorage.getInMemoryCache();
+        expect(Object.keys(inMemoryCache.accessTokens).length).toBe(0);
+    });
+
+    it("setInMemoryCache() and getCache()", () => {
+        const nodeStorage = new Storage(logger);
+        nodeStorage.setInMemoryCache(inMemoryCache);
+
+        const cache = nodeStorage.getCache();
+        const accountKey = 'uid.utid-login.microsoftonline.com-microsoft';
+        const account = cache[accountKey] as AccountEntity;
+        expect(account).toBeInstanceOf(AccountEntity);
+        expect(account.clientInfo).toBe('eyJ1aWQiOiJ1aWQiLCAidXRpZCI6InV0aWQifQ==');
+    });
+
+    it('setInMemoryCache() and getInMemoryCache() tests', () => {
+        const nodeStorage = new Storage(logger);
+        nodeStorage.setInMemoryCache(inMemoryCache);
+
+        const cache = nodeStorage.getCache();
+        const accountKey = 'uid.utid-login.microsoftonline.com-microsoft';
+        const account = cache[accountKey] as AccountEntity;
+        expect(account).toBeInstanceOf(AccountEntity);
+        expect(account.clientInfo).toBe(
+            'eyJ1aWQiOiJ1aWQiLCAidXRpZCI6InV0aWQifQ=='
+        );
+
+        const newInMemoryCache = nodeStorage.getInMemoryCache();
+        expect(newInMemoryCache.accounts[accountKey]).toEqual(account);
+    });
+
+    it('setItem() and getItem() tests', () => {
```

Comment:
Can we also specify in the titles which part they are testing? e.g. Account, AccessToken, etc?

Reply:
well I distributed the data types to be random, will add.